apps/etl-func/src/adapters/pdv-tokenizer/pdv-tokenizer-client.ts

```diff
@@ -0,0 +1,57 @@
+import { TokenizerClient } from "@/domain/interfaces/tokenizer.js";
+
+import { piiResourceSchema } from "./pii-resource.js";
+import { problemSchema } from "./problem.js";
+import { tokenResourceSchema } from "./token-resource.js";
+
+const delay = (ms: number) => new Promise((resolve) => setTimeout(resolve, ms));
+
+export default class PDVTokenizerClient implements TokenizerClient {
+  #apiKey: string;
+  #basePath = "tokenizer/v1";
+  #baseUrl = "https://api.tokenizer.pdv.pagopa.it/";
+
+  constructor(apiKey: string) {
+    if (!apiKey) throw new Error("Api key is required");
```

Comment:
Is this check needed? We are accepting `string` in input and not `string | undefined`, also the Api key env var is required

Reply:
Even if the parameter is a string you can pass an empty string, this check avoids that